extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/OpenTelemetryProcessor.java

```diff
@@ -81,14 +97,42 @@ public List<InterceptorBinding> getAdditionalBindings() {
     }
 
     @BuildStep
-    void transformWithSpan(
-            BuildProducer<AnnotationsTransformerBuildItem> annotationsTransformer) {
+    void transformWithSpan(BuildProducer<AnnotationsTransformerBuildItem> annotationsTransformer) {
+
+        // Transform deprecated annotation into new one
+        annotationsTransformer.produce(new AnnotationsTransformerBuildItem(new AnnotationsTransformer() {
+            @Override
+            public boolean appliesTo(AnnotationTarget.Kind kind) {
+                return AnnotationsTransformer.super.appliesTo(AnnotationTarget.Kind.METHOD);
```

Comment:
That will still always return `true`. I think you need `return kind == AnnotationTarget.Kind.METHOD;`.

Reply:
Fixed. Sorry.